iModelCore/ecobjects/src/ECcontext.cpp

```diff
@@ -357,6 +359,15 @@ struct ECSchemaReadContextBackedInstanceReadContext: public ECInstanceReadContex
 
         return schema.get();
         }
+
+    /*---------------------------------------------------------------------------------**//**
+    * @bsimethod
+    +---------------+---------------+---------------+---------------+---------------+------*/
+    virtual bool IsSchemaToBePruned(SchemaKeyCR key) const
+        {
+        const auto& schemasToPrune = m_schemaReadContext.GetSchemasToPrune();
+        return std::find(schemasToPrune.begin(), schemasToPrune.end(), key.GetName()) != schemasToPrune.end();

```

Comment:
I think we will want to match by name only, and not by full schema key

Reply:
This is matching by name only `key.GetName()`